File: content/src/main/java/org/runestar/cache/content/EnumDefinition.java
package org.runestar.cache.content;

import java.nio.ByteBuffer;

public class EnumDefinition {

    public int[] intVals = null;

    public char keyType = 0;

    public char valType = 0;

    public String defaultString = null;

    public int defaultInt = 0;

    public int size = 0;

    public int[] keys = null;

    public String[] stringVals = null;

    public void read(ByteBuffer buffer) {
        while (true) {
            int opcode = Byte.toUnsignedInt(buffer.get());
            switch (opcode) {
                case 0:
                    return;
                case 1:
                    keyType = (char) buffer.get();
                    break;
                case 2:
                    valType = (char) buffer.get();
                    break;
                case 3:
                    defaultString = Bytes.readString(buffer);
                    break;
                case 4:
                    defaultInt = buffer.getInt();
                    break;
                case 5:
                    size = Short.toUnsignedInt(buffer.getShort());
                    keys = new int[size];
                    stringVals = new String[size];
                    for (int i = 0; i < size; i++) {
                        keys[i] = buffer.getInt();
                        stringVals[i] = Bytes.readString(buffer);
                    }
                    break;
                case 6:
                    size = Short.toUnsignedInt(buffer.getShort());
                    keys = new int[size];
                    intVals = new int[size];
                    for (int i = 0; i < size; i++) {
                        keys[i] = buffer.getInt();
                        intVals[i] = buffer.getInt();
                    }
                    break;
                default:
                    throw new UnsupportedOperationException(String.valueOf(opcode));
            }
        }
    }
}

File: src/main/java/com/runesuite/cache/content/def/EnumDefinition.kt
//package com.runesuite.cache.content.def
//
//import com.hunterwb.kxtra.nettybuffer.bytebuf.readNullTerminatedString
//import com.runesuite.general.CHARSET
//import io.netty.buffer.ByteBuf
//
//class EnumDefinition : CacheDefinition() {
//
//    var intVals: IntArray? = null
//    var keyType: Char = ' '
//    var valType: Char = ' '
//    var defaultString = "null"
//    var defaultInt: Int = 0
//    var keys: IntArray? = null
//    var stringVals: Array<String>? = null
//
//    override fun read(buffer: ByteBuf) {
//        while (true) {
//            val opcode = buffer.readUnsignedByte().toInt()
//            when (opcode) {
//                0 -> return
//                1 -> keyType = buffer.readUnsignedByte().toChar()
//                2 -> valType = buffer.readUnsignedByte().toChar()
//                3 -> defaultString = buffer.readNullTerminatedString(CHARSET)
//                4 -> defaultInt = buffer.readInt()
//                5 -> {
//                    val length = buffer.readUnsignedShort()
//                    keys = IntArray(length)
//                    stringVals = Array(length) { "" }
//                    for (i in 0 until length) {
//                        keys!![i] = buffer.readInt()
//                        stringVals!![i] = buffer.readNullTerminatedString(CHARSET)
//                    }
//                }
//                6 -> {
//                    val length = buffer.readUnsignedShort()
//                    keys = IntArray(length)
//                    intVals = IntArray(length)
//                    for (i in 0 until length) {
//                        keys!![i] = buffer.readInt()
//                        intVals!![i] = buffer.readInt()
//                    }
//                }
//                else -> error(opcode)
//            }
//        }
//    }
//
//    override fun toString(): String {
//        return "EnumDefinition(keyType=$keyType, valType=$valType, defaultString=$defaultString, stringVals=${stringVals?.contentToString()})"
//    }
//}